server/src/main/java/io/deephaven/server/console/completer/PythonAutoCompleteObserver.java

```diff
@@ -0,0 +1,222 @@
+                    item.setStart(start);
+                    item.setLabel(completionName);
+                    item.setLength(completionName.length());
+                    range.getStartBuilder().setLine(pos.getLine()).setCharacter(start);
+                    range.getEndBuilder().setLine(pos.getLine()).setCharacter(start + completionName.length());
+                    item.setInsertTextFormat(2);
+                    if (completionName.startsWith("__")) {
+                        completionResults__.add(item);
+                    } else if (completionName.startsWith("_")) {
+                        completionResults_.add(item);
+                    } else {
+                        completionResults.add(item);
+                    }
+                }
+                int sortPos = 0;
+                List<CompletionItem> finalItems = new ArrayList<>();
+                for (CompletionItem.Builder res : completionResults) {
+                    res.setSortText(ChunkerCompleter.sortable(sortPos++));
+                    finalItems.add(res.build());
+                }
+                for (CompletionItem.Builder res : completionResults_) {
+                    res.setSortText(ChunkerCompleter.sortable(sortPos++));
+                    finalItems.add(res.build());
+                }
+                for (CompletionItem.Builder res : completionResults__) {
+                    res.setSortText(ChunkerCompleter.sortable(sortPos++));
+                    finalItems.add(res.build());
+                }
+
+                final GetCompletionItemsResponse builtItems = GetCompletionItemsResponse.newBuilder()
+                        .setSuccess(true)
+                        .setRequestId(request.getRequestId())
+                        .addAllItems(finalItems)
+                        .build();
+
+                safelyExecuteLocked(responseObserver,
+                        () -> responseObserver.onNext(AutoCompleteResponse.newBuilder()
+                                .setCompletionItems(builtItems)
+                                .build()));
+                String totalNano = Long.toString(System.nanoTime() - startNano);
+                // lets track how long completions take, as it's known that some
+                // modules like numpy can cause slow completion, and we'll want to know what was causing them
+                log.info().append("Jedi completions took ")
+                        .append(
+                                totalNano.length() > 6 ?
+                                        totalNano.substring(0, totalNano.length() - 6) :
+                                        "0"
+                        )
+                        .append('.')
+                        .append(totalNano.substring(6, Math.min(8, totalNano.length())))
+                        .append("ms").endl();
+
+            } catch (Throwable e) {
+                log.error().append("Jedi completion failure ").append(e).endl();
```

Comment:
this catch should notify client of failure too?

Reply:
just removed the catch. I don't think I need to eat Throwable here, just let outer catch have it